internal/cmd/skupper/link/kube/link_create.go

```diff
@@ -0,0 +1,200 @@
+
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	numberValidator := validator.NewNumberValidator()
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	//Validate if there is already a site defined in the namespace
+	siteList, _ := cmd.Client.Sites(cmd.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if siteList != nil && len(siteList.Items) == 0 {
+		validationErrors = append(validationErrors, fmt.Errorf("there is no skupper site in this namespace"))
+	}
+
+	if len(args) == 0 || args[0] == "" {
+		validationErrors = append(validationErrors, fmt.Errorf("link name must not be empty"))
+	} else if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command."))
+	} else {
+		cmd.linkName = args[0]
+
+		ok, err := resourceStringValidator.Evaluate(cmd.linkName)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("link name is not valid: %s", err))
+		}
+	}
+
+	if cmd.flags.tlsSecret != "" {
+		secret, err := cmd.KubeClient.CoreV1().Secrets(cmd.Namespace).Get(context.TODO(), cmd.flags.tlsSecret, metav1.GetOptions{})
+		if secret == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("the TLS secret %q is not available in the namespace: %s", cmd.flags.tlsSecret, err))
+		}
+	}
+
+	selectedCost, err := strconv.Atoi(cmd.flags.cost)
+	if err != nil {
+		validationErrors = append(validationErrors, fmt.Errorf("link cost is not valid: %s", err))
+	}
+	ok, err := numberValidator.Evaluate(selectedCost)
+	if !ok {
+		validationErrors = append(validationErrors, fmt.Errorf("link cost is not valid: %s", err))
+	}
+
+	if cmd.flags.output != "" {
+		ok, err := outputTypeValidator.Evaluate(cmd.flags.output)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
+		}
+	}
+
+	return validationErrors
+}
+
+func (cmd *CmdLinkCreate) InputToOptions() {
+
+	cmd.cost, _ = strconv.Atoi(cmd.flags.cost)
+	cmd.tlsSecret = cmd.flags.tlsSecret
+	cmd.output = cmd.flags.output
+
+}
+
+func (cmd *CmdLinkCreate) Run() error {
+
+	resource := v1alpha1.Link{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "skupper.io/v1alpha1",
+			Kind:       "Link",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cmd.linkName,
+			Namespace: cmd.Namespace,
+		},
+		Spec: v1alpha1.LinkSpec{
+			TlsCredentials: cmd.tlsSecret,
+			Cost:           cmd.cost,
+		},
+	}
+
+	if cmd.output != "" {
+		encodedOutput, err := utils.Encode(cmd.output, resource)
+		fmt.Println(encodedOutput)
+
+		return err
+
+	} else {
+		_, err := cmd.Client.Links(cmd.Namespace).Create(context.TODO(), &resource, metav1.CreateOptions{})
+		return err
+	}
+
+}
+
+func (cmd *CmdLinkCreate) WaitUntilReady() error {
+
+	// the site resource was not created
+	if cmd.output != "" {
+		return nil
+	}
+
+	err := utils.NewSpinner("Waiting for status...", 5, func() error {
```

Comment:
`link_create.go` is being deleted, do you refer to `link_export.go`?

Reply:
is there a new refdog with "link export" vs "link create" command?

Reply:
No, that is why I have added `new` in the description of the PR. The `link create` command was not helping the user to create a link because they needed to specify the endpoint information from the site target manually. So after discussing it with @ssorj and @grs the link export command will generate a yaml file with all the resources needed to create a link. Eventually, the refdog documentation will make reference to this new command.